workflows/prognostic_c48_run/runtime/transformers/tendency_prescriber.py

```diff
@@ -27,55 +24,56 @@ class TendencyPrescriberConfig:
         mapper_config: configuration of mapper used to load tendency data.
         variables: mapping from state name to name of corresponding tendency in
             provided mapper. For example: {"air_temperature": "fine_res_Q1"}.
+        reference_initial_time: if time interpolating, time of first point in dataset
+        reference_frequency_seconds: time frequency of dataset
         limit_quantiles: mapping of "upper" and "lower" keys to quantile specifiers
             for limiting extremes in the Q1, Q2 dataset
     """
 
     mapper_config: loaders.MapperConfig
     variables: Mapping[str, str]
+    reference_initial_time: Optional[str] = None
+    reference_frequency_seconds: float = 900
     limit_quantiles: Optional[Mapping[str, float]] = None
 
 
 @dataclasses.dataclass
 class TendencyPrescriber:
     """Wrap a Step function and prescribe certain tendencies."""
 
-    config: TendencyPrescriberConfig
     state: DerivedFV3State
     communicator: pace.util.CubedSphereCommunicator
     timestep: float
+    variables: Mapping[str, str]
+    mapper_func: Callable[[cftime.DatetimeJulian], State]
+    limit_quantiles: Optional[Mapping[str, float]] = None
```

Comment:
While you are extracting things from this class you could  remove the limiter logic too. This could be bundled into the `mapper_func` using a easily tested function like this:
```
def limit_mapper_func(limiter: DatasetQuantileLimiter, mapper_func):
    def limited_func(time):
        return limiter.transform(mapper_func(time)) 
    return limited_func
```

This will allow you to delete two methods, the optional `self._limiter` attribute + corresponding if statements, and remove one argument from the constructor. You can initialize the `limiter` close to where you initialize the mapper_func.

I suppose this optional since the limiting logic isn't new, but I think it would make this class align much better with the Single Responsibility Principle. The whole purpose of the class is just to execute mapper_func on the root rank and scatter it to the others.

Reply:
Agree that separation is desirable, will do